### DISCUSSION TITLE
Get name of livewire view on a blade component?

### DISCUSSION THREAD

**Discussion by theoreaves** [livewire/livewire#4764] [1. Help]
So, this may be an odd request.  But:

I'm working on a x-componet, and I want to know the name of the livewire blade the componet is getting included from so that I can reach into a database table to pull out some information as to the requirements of the field.  But I'm at a loss on how to get this data.  Really, it can be anything that is unique to that screen, but the same for anyone when they go to that screen.

**Comment by silnarm** [comment]
I have to say it... Yeah, it's odd.
I'm not asking, please don't tell me.
As to how, a standard arg to the blade component should suffice shouldn't it?
```
    <x-test livewire-parent="foo-component" ... />
```
You have to provide it for each instance, but you're right there in the livewire component's blade, so that doesn't seem a great burden (prone to copy-paste errors though).

**Reply by theoreaves** [reply]
yeah, that is what I'm starting to lean towards. I just hoped that there may be some magic variable I didn't know about.

So, you didn't ask but... my reason is so that field requirements could be stored in the database and be different based on each tenant i the system.  It may be a bit of a reach, but I'm going to give it a try anyway...

**Reply by silnarm** [reply]
Legit scenario, architecture seems a little convoluted though? Still not sure I get it completely, but it seems the Livewire component should know the 'tenant' and could maybe have the field requirements already to pass onto embedded Blade components?

Regarding some auto-magic way to set the parent, the blade context gets in the way, I think the best you could do might be to have a new Component base class:
```
class ComponentBase extends \Livewire\Component
{
    public $className = '';
    public function __construct() {
        parent::__construct();
        $this->className = static::class;
    }
}
```
Which adds a public property with the derived class name.
Then you can eliminate the need to specify the parent when including the blade component, so long as your Livewire component extends ComponentBase instead of Component they can all just do `<x-test livewire-parent="{{$className}}" ... />`

**Comment by joshhanley** [comment]
@theoreaves in your Livewire blade view, you should have access to `$this` which will be bound to the Livewire instance, so you should be able to inspect that to find out what it's class name is and pass that into your blade component.

Hope this helps!
